src/Microsoft.ML.DataView/DataViewType.cs

```diff
@@ -37,11 +37,25 @@ private protected DataViewType(Type rawType)
         /// </remarks>
         public Type RawType { get; }
 
-        // IEquatable<T> interface recommends also to override base class implementations of
-        // Object.Equals(Object) and GetHashCode. In classes below where Equals(ColumnType other)
-        // is effectively a referencial comparison, there is no need to override base class implementations
-        // of Object.Equals(Object) (and GetHashCode) since its also a referencial comparison.
+        /// <summary>
+        /// Return <see langword="true"/> if <see langword="this"/> is equivalent to <paramref name="other"/> and <see langword="false"/> otherwise.
+        /// </summary>
+        /// <param name="other">Another <see cref="DataViewType"/> to be compared with <see langword="this"/>.</param>
         public abstract bool Equals(DataViewType other);
+
+        /// <summary>
+        /// Produce the hashing code of <see langword="this"/>. It's the implementation of <see cref="GetHashCode()"/>.
+        /// </summary>
+        public abstract int GetDataViewTypeHashCode();
```

Comment:
This is actually a breaking change and shouldn't be checked in. You can't add new `abstract` methods to classes that have already shipped.

Reply:
@artidoro @ericstj - this should be caught by ApiCompat, correct? Is that working in this repo?

Reply:
Should be https://github.com/dotnet/arcade/blob/767154ac5bb8715ab1357c63f346244de6d4aa6b/src/Microsoft.DotNet.ApiCompat/Rules/Compat/CannotAddAbstractMembers.cs

I can’t see the previous build result, did APICompat run?

Reply:
Yes, ApiCompat does run, I tested it locally.

This appears to be a bug in the rule.  Here is the scenario:

# V1
```C#
public abstract class InternalConstructorBase
{
    private protected InternalConstructorBase()
    { }
}

public abstract class IntermediateBase
{
    protected InternalConstructorBase()
    { }
}
```

# V2 - add an abstract member to `InternalConstructorBase`
```C#
public abstract class InternalConstructorBase
{
    private protected InternalConstructorBase()
    { }

    public abstract void NewMember();
}

public abstract class IntermediateBase
{
    protected InternalConstructorBase()
    { }
}
```

Here, `InternalConstructorBase` is "effectively sealed" so it sneaks by the rule in 

https://github.com/dotnet/arcade/blob/767154ac5bb8715ab1357c63f346244de6d4aa6b/src/Microsoft.DotNet.ApiCompat/Rules/Compat/CannotAddAbstractMembers.cs#L26-L27

But, `IntermediateBase` now has a new `abstract` member, from its base class.

I've opened https://github.com/dotnet/arcade/issues/2885 for it.